### PR TITLE
Added more documentation

### PR DESCRIPTION
**Pull request by jpark052**
Added more doc including some relevant research papers, and modified the format a little bit as well. I added a bunch of issues as well. Thanks!

### CODE REVIEW

**Review by Syyreign — APPROVED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -27,20 +46,20 @@ Retro chiptune music is starting to make a resurgence with the popularity of man
 ---
 Mainly Rust, and possibly Python as well
 
-## Possible Function Implemenations
+## Expected Functions:
 ---
 * MIDI Input
 * Visualization (waveform)
 * GUI
 * Pass songsusing import
```
Comment (Quayvid) [inline README.md]: Should add a space between "songs" and "using"

File: README.md [hunk 2]
```diff
@@ -27,20 +46,20 @@ Retro chiptune music is starting to make a resurgence with the popularity of man
...
 * MIDI Input
 * Visualization (waveform)
 * GUI
 * Pass songsusing import 
 
-## Good to have:
+## Potential Functions:
 ---
 * Filters?
 * Entrie chip (real code)
```
Comment (Quayvid) [inline README.md:60]: Dunno if that's supposed to be "Entry chip" or if how that is spelled

File: README.md [hunk 3]
```diff
@@ -4,6 +4,25 @@
 ---
 Retro chiptune music is starting to make a resurgence with the popularity of many retro indie games like Hades and Celeste. One of the reasons why it's gaining its popularity again is because it resonates with a lot of people; it brings back nostalgia and fond memory associated with it. However, a majority of music produced for these games rarely use actual hardware that made the 8-bit and 16-bit music charming. The purpose of this project is to replicate a sound chip from the NES (Nintendo Entertainment System) using Rust, and bring back those memories to the users.
 
+## Milestones
+---
+### Milestone 1 (worst case scenario):
+>* Basic audio generator that can produce 8-bit sound based on user's keyboard input.
+
+### Milestone 2 (expected):
+>* The program can accept MIDI file as an input, and convert it to chiptune sound. It should be able to handle multiple channels as well.
```
Comment (Quayvid) [inline README.md]: Should probably change the first sentence to "The program can accept a MIDI file as input, and convert it to chiptune sound."

File: README.md [hunk 4]
```diff
@@ -4,6 +4,25 @@
...
+Various MIDI files should be used for testing this function, such as relatively simple classical music file with a single channel, and more complex pop song with multiple channels. <br /><br />
```
Comment (Quayvid) [inline README.md]: Part of the first sentence should be changed to "such as a relatively simple classical music file with a single channel, and more complex pop songs with multiple channels."

File: README.md [hunk 5]
```diff
@@ -4,6 +4,25 @@
...
+>* Graphical User Interface is implemented with features such as play/pause, volume control, and MIDI import, export function.<br /><br />
+>* Animation of waveform is displayed as well when music is played (something like old Windows' winamp visualization)
```
Comment (Quayvid) [inline README.md]: Change "waveform" to "waveforms" probably

**Review by Quayvid — COMMENTED** [review]
Went over some stuff in the Progress Report that should be changed (mainly wording and grammar)

File: README.md (hunk 2 above)
Comment (jpark052) [reply README.md:60]: Yeah not sure about that either tbh, just gonna leave it as is

**Review by Quayvid — APPROVED** [review]
The progress report looks good to me, think we can merge it unless anyone else thinks more should be added to it or other edits need to be made

File: README.md [hunk 6]
```diff
@@ -1,40 +1,75 @@
...
+We could test this function by comparing the visualization with other pre-existing services.
+
+### Milestone 3 (advanced):
+>* An equalizer is added and users can use it to modify the sound, like base boost, treble boost, etc.
+This also can be tested by utilizing other services, such as spotify's equalizer function and comparing the difference 
+in quality.
+
+## Contributors
+---
+[Cy Chung](https://github.com/crschung), [Jae Park](https://github.com/jpark052), [Syyreign](https://github.com/Syyreign), [David Kim](https://github.com/Quayvid), [Ethan Slogotski](https://github.com/eman1003), [Francis German](francisgerman70), [Skylar Buck](https://github.com/skylarbuck)
```
Comment (Syyreign) [inline README.md]: Would you be able to add my name? It's Spencer Hart.

**Review by Syyreign — APPROVED** [review]
Looking good!

**Pull request merged**